pkg/daemon/daemon.go

```diff
@@ -785,10 +785,37 @@ func (dn *Daemon) getDrainLock(ctx context.Context, done chan bool) {
 	})
 }
 
+// isMachineConfigPoolResourceRegistered checks to see if the MachineConfigPool API Resource
+// is registered in the Openshift cluster
+func isMachineConfigPoolResourceRegistered(c *mcclientset.Clientset) (bool, error) {
+	machineConfigPoolResourceName := "machineconfigpool"
+	availableAPIs, err := c.ServerResourcesForGroupVersion(mcfgv1.GroupVersion.String())
+	if err != nil && !errors.IsNotFound(err) {
+		return false, err
+	}
+	if availableAPIs != nil {
+		for _, api := range availableAPIs.APIResources {
+			if api.SingularName == machineConfigPoolResourceName {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
 func (dn *Daemon) pauseMCP() error {
 	glog.Info("pauseMCP(): pausing MCP")
 	var err error
-
+	isRegistered, err := isMachineConfigPoolResourceRegistered(dn.mcClient)
+	if err != nil {
+		return err
+	}
+	// In hypershift: there are no machineConfigPool resources
+	// if the MachineConfigPool resource is not registered in API there is no need to pause it
+	if !isRegistered {
+		glog.Info("pauseMCP(): MCP resource not registered")
```

Comment:
I agree that there is no need to pause it, but this isn't the only dependency on MCP that you need to properly handle to support sriov network operator in Hypershift.

Reply:
right I do think there are potentially a couple others that would need a similar check on "machineConfig" but was going to break those up into smaller prs